From 511cd57de795d7c683c30427130ba985b221a024 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 13 Feb 2018 09:33:41 -0800
Subject: [PATCH 0270/1852] drm: xlnx: zynqmp: disp: Use the device node for
 layers

commit 5cc922294435407a8e51516843e29763b88a9848 from
https://github.com/Xilinx/linux-xlnx.git

Without Xilinx bridge support, there's no need for child nodes
for layers. Use the main device node for child layers.
Leave the layer node member so that adding bridge support becomes
easier.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index 5f527671b88b..f46be97d3fc8 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -2014,17 +2014,12 @@ static int zynqmp_disp_layer_create(struct zynqmp_disp *disp)
 	int ret;
 
 	for (i = 0; i < ZYNQMP_DISP_NUM_LAYERS; i++) {
-		char temp[16];
-
 		layer = &disp->layers[i];
 		layer->id = i;
 		layer->offset = i * 4;
 		layer->other = &disp->layers[!i];
 		layer->num_chan = num_chans[i];
-		snprintf(temp, sizeof(temp), "%s-layer", dma_name[i]);
-		layer->of_node = of_get_child_by_name(disp->dev->of_node, temp);
-		if (!layer->of_node)
-			goto err;
+		layer->of_node = disp->dev->of_node;
 		ret = zynqmp_disp_layer_request_dma(disp, layer, dma_name[i]);
 		if (ret)
 			goto err;
-- 
2.31.1

